app/inline-scripts/load-ads.js

```diff
@@ -0,0 +1,44 @@
+(function () {
+	var getterAdsQueue = [];
+	var isAdsListening = false;
+	var adsLoaded = false;
+
+	window.getInstantGlobal('wgSitewideDisableAdsOnMercury', (wgSitewideDisableAdsOnMercury) => {
+		if (wgSitewideDisableAdsOnMercury || (new URL(document.location)).searchParams.get('noexternals')) {
+			return;
+		}
+
+		const wikiVariables = window.M.getFromHeadDataStore('wikiVariables');
+
+		window.M.loadScript(
+			wikiVariables.cdnRootUrl + '/__am/' + wikiVariables.cacheBuster + '/groups/-/mercury_ads_js',
+			true,
+			function () {
+				window.document.dispatchEvent(new Event('adsScriptLoaded'));
+			}
+		);
+	});
+
+	window.waitForAds = function (callback) {
+		function onAdsLoaded() {
```

Comment:
this will create as many functions as calls to waitForAds - intentional?

Reply:
Nice catch. Fixed and also simplified the code a little bit.